### PR TITLE
feat: avatar texture

### PR DESCRIPTION
**Pull request by pravusjif**
* Added avatar texture option in Material component.
* Separated Texture to be a common message and not something inside Material.
* Now every texture usage (`TextureUnion`) should be declared in the following way:

```
  const mat = Material.create(entity, {
    texture: { 
      texture: {
        src: "images/scene-thumbnail.png"
      },
      avatarTexture: undefined
      
      // texture: undefined,
      // avatarTexture: {
      //   userId: "0x51777c0b8DBA8B4dfE8a1c3d0a1eDAA5B139B4e0" 
      // }
    }
  })
```

### CODE REVIEW

**Review by leanmendoza — DISMISSED** [review]
LGTM! Before the release of ECS7, we can evaluate if the `albedoTexture` field should be a deeper one as `albedoTexture.avatarTexture` and `albedoTexture.texture` or keep this approach with a root exposing of the `oneof` (like it already has the MeshRenderer/MeshCollider).

**Review by menduz — CHANGES_REQUESTED** [review]

**Comment by menduz** [comment]
Cross posting from https://github.com/decentraland/unity-renderer/pull/3235#discussion_r999832792

---

Here are references to this `oneof` occurrence
- https://github.com/decentraland/protocol/blob/main/proto/decentraland/sdk/components/mesh_renderer.proto#L24-L29

To map this to a concept, ideally we have a single `oneof` for each interface like this


```proto
message ITexture {
  oneof tex {
    ImageTexture image = 1;
    AvatarTexture avatar = 2;
    GltfTexture avatar = 3;
    // ResourceManagerTexture from_resource = 4;
  }
}

message Material {
  ITexture albedo_texture = 1;
  ITexture alpha_texture = 2;
}
```

And then in the implementation, the component decides how to get the specific (real) resource based on the union type like this

```ts

function getResourceFromTexture(protoTexture: PB.ITexture) {
  switch (......)
}

function updateMaterial(materialModel) {
  realMaterial.texture = getResourceFromTexture(materialModel.albedoTexture)
  realMaterial.alphaTexture = getResourceFromTexture(materialModel.alphaTexture)
}
```

That is extremely important to enable protocol extensibility and new kinds of textures. We can see clearly that once we need VideoTexture, it can be added to the `oneof`

```diff
 message ITexture {
   oneof tex {
     ImageTexture image = 1;
     AvatarTexture avatar = 2;
     GltfTexture avatar = 3;
+    VideoTexture video = 4;
   }
 }
```

File: proto/decentraland/common/texture.proto [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+message AvatarTexture {
+  string user_id = 1;
+  optional TextureWrapMode wrap_mode = 2; // default = TextureWrapMode.Clamp
+  optional TextureFilterMode filter_mode = 3; // default = FilterMode.Bilinear
+}
+
+message Texture {
+    oneof texture {
+      SRCTexture src_texture = 1; // default = null
```
Comment (menduz) [inline proto/decentraland/common/texture.proto:31]:
```suggestion
      SrcTexture src_texture = 1; // default = null
```

File: proto/decentraland/sdk/components/material.proto [hunk 2]
```diff
@@ -29,19 +17,14 @@ enum MaterialTransparencyMode {
...
+  
   optional float alpha_test = 2; // default = 0.5. range value: from 0 to 1
   optional bool cast_shadows = 3; // default =  true
 
-  optional Texture alpha_texture = 4; // default = null
-  optional Texture emissive_texture = 5; // default = null
-  optional Texture bump_texture = 6; // default = null
+  optional decentraland.common.SRCTexture alpha_texture = 4; // default = null
+  optional decentraland.common.SRCTexture emissive_texture = 5; // default = null
+  optional decentraland.common.SRCTexture bump_texture = 6; // default = null
```
Comment (menduz) [inline proto/decentraland/sdk/components/material.proto:27]:
```suggestion
  optional decentraland.common.Texture alpha_texture = 4; // default = null
  optional decentraland.common.Texture emissive_texture = 5; // default = null
  optional decentraland.common.Texture bump_texture = 6; // default = null
```

**Comment by pravusjif** [comment]
@menduz I implemented the requested changes.

Be aware that now the SDK nomenclature is a bit uglier:

```
  const mat = Material.create(entity, {
    texture: { 
      texture: {
        src: "images/scene-thumbnail.png"
      },
      avatarTexture: undefined
      
      // texture: undefined,
      // avatarTexture: {
      //   userId: "0x51777c0b8DBA8B4dfE8a1c3d0a1eDAA5B139B4e0" 
      // }
    }
  })
```

File: proto/decentraland/sdk/components/material.proto [hunk 3]
```diff
@@ -29,19 +17,14 @@ enum MaterialTransparencyMode {
...
+  
   optional float alpha_test = 2; // default = 0.5. range value: from 0 to 1
   optional bool cast_shadows = 3; // default =  true
 
-  optional Texture alpha_texture = 4; // default = null
-  optional Texture emissive_texture = 5; // default = null
-  optional Texture bump_texture = 6; // default = null
+  optional decentraland.common.Texture alpha_texture = 4; // default = null
+  optional decentraland.common.Texture emissive_texture = 5; // default = null
+  optional decentraland.common.Texture bump_texture = 6; // default = null
```
Comment (menduz) [inline proto/decentraland/sdk/components/material.proto:27]: This three should also be TextureUnion to enable extensibility

**Review by menduz — CHANGES_REQUESTED** [review]

**Review by menduz — APPROVED** [review]

**Pull request merged**